src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +355,118 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
         assertEquals(0, comments.size());
     }
 
+    @Test
+    public void testIsNameVisibleToUser() {
+        FeedbackResponseCommentAttributes comment = dataBundle.feedbackResponseComments.get("comment1FromT1C1ToR1Q1S1C1");
+        FeedbackResponseAttributes relatedResponse = dataBundle.feedbackResponses.get("response1ForQ1S1C1");
+        CourseRoster roster = new CourseRoster(new ArrayList<>(dataBundle.students.values()),
+                new ArrayList<>(dataBundle.instructors.values()));
+
+        ______TS("success: the list that comment giver's name shown to is null; always return true");
+        comment.showGiverNameTo = null;
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "", roster));
+
+        ______TS("success: comment's visibility follows feedback question; always return ture");
+        comment.isVisibilityFollowingFeedbackQuestion = true;
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "", roster));
+
+        ______TS("success: comment is always visible to its giver");
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "instructor1@course1.tmt", roster));
+
+        comment = dataBundle.feedbackResponseComments.get("comment1FromT1C1ToR1Q1S2C2");
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "instructor3@course.tmt", roster));
+
+        // test user's participant type is in the list that comment giver's name is shown to
+        comment = dataBundle.feedbackResponseComments.get("comment1FromT1C1ToR1Q2S1C1");
+        relatedResponse = dataBundle.feedbackResponses.get("response1ForQ2S1C1");
+        relatedResponse.giver = "student1InCourse1@gmail.tmt";
+
+        ______TS("success: comment is only visible to instructors");
+        comment.commentGiver = "student6@course1.tmt";
+        comment.commentGiverType = STUDENTS;
+        comment.showGiverNameTo = Arrays.asList(INSTRUCTORS);
+        assertTrue(frcLogic.isNameVisibleToUser(comment, relatedResponse, "instructor2@course1.tmt", roster));
```

Comment:
e.g "instructor2@course1.tmt" can be a constant as its being reused in a lot of places